linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -647,6 +657,39 @@ final class Emitter(config: Emitter.Config) {
       _importsCache
     }
 
+    def getOrComputeTopLevelExports(topLevelExports: List[LinkedTopLevelExport])(
+        compute: => WithGlobals[List[js.Tree]]): WithGlobals[List[js.Tree]] = {
+
+      _cacheUsed = true
+
+      if (!sameTopLevelExports(topLevelExports, _lastTopLevelExports)) {
+        _topLevelExportsCache = compute
+        _lastTopLevelExports = topLevelExports
+      }
+      _topLevelExportsCache
+    }
+
+    private def sameTopLevelExports(tles1: List[LinkedTopLevelExport], tles2: List[LinkedTopLevelExport]): Boolean = {
+      import org.scalajs.ir.Trees._
+
```

Comment:
Maybe drop a comment in this method, that the target module is the same due to the context. (It threw me off for a second why to tles with different target modules would be considered equal here).